moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -815,7 +813,10 @@ bool ompl_interface::ModelBasedPlanningContext::solve(double timeout, unsigned i
           for (unsigned int i = 0; i < max_planning_threads_; ++i)
             ompl_parallel_plan_.addPlanner(ompl::tools::SelfConfig::getDefaultPlanner(ompl_simple_setup_->getGoal()));
         bool r = ompl_parallel_plan_.solve(ptc, 1, count, hybridize_) == ompl::base::PlannerStatus::EXACT_SOLUTION;
-        result = result && r;
+        // Was this latest call successful too?
+        result.val = (result.val == moveit_msgs::MoveItErrorCodes::SUCCESS && r) ?
+                         moveit_msgs::MoveItErrorCodes::SUCCESS :
+                         moveit_msgs::MoveItErrorCodes::FAILURE;
```

Comment:
I agree that this correctly translates the original code's idea.
However, it becomes more difficult to comprehend. As the actual OMPL error code isn't used anymore (which should be fixed IMO), I suggest a simplification:
https://github.com/AndyZe/moveit/pull/3

I think there is also a logical error in the old code:
Running multiple planning attempts in parallel (or sequence as here) is done to increase the chance of planning success, isn't it? Thus, only return success if all attempts succeeded doesn't make sense to me.

Reply:
I addressed the remaining issues here:
https://github.com/ubi-agni/moveit/compare/pr/AndyZe/3257...ubi-agni:moveit:fix-parallel-planning

I suggest merging this PR with the proposed changes (https://github.com/AndyZe/moveit/pull/3) and subsequently handle the code cleanup in a separate PR.